### ISSUE TITLE
A question on ResidualWrapper

### ISSUE THREAD

**Issue by yuyujunjun** [una-dinosauria/human-motion-prediction#68]
Hi @una-dinosauria ,
I have a question on ResidualWrapper. The Encoder cell is warpped by residual warpper just like the Decoder cell. In #42 , you explained that:

>  it is the case for the extra outputs of the encoder -- they do not affect the loss.

However, I think the residual warpper will affect the encoder's input. The final input is the original input, which read from dataset, plus the LinearWarpper's output at previous timestep. 

I tested the pytorch version. (Because I am not familiar with tensorflow. If I am wrong please correct me, thanks! ) 

First I use the encoder just like this: get the input and update the hidden state

```
    for i in range(self.source_seq_len-1):
        enc_input=encoder_inputs[i]
        state = self.cell(enc_input, state)#RNN
        state = F.dropout(state, self.dropout, training=self.training)#dropout=0.0
```

But the result is wrong:

![image](https://user-images.githubusercontent.com/13601549/114078854-093aaf00-98dc-11eb-93a3-16cfc68729c9.png)

And then I tried add the linearWarpper's output to the original input:

```
    for i in range(self.source_seq_len-1):
        enc_input=encoder_inputs[i]
        if output is not None:
           enc_input=enc_input+output
        state = self.cell(enc_input, state)#RNN
        state = F.dropout(state, self.dropout, training=self.training)#dropout=0.0
        output=self.fc1(state)
```

The result is also not the same as the tensorflow version, but it looks fine finally.

![image](https://user-images.githubusercontent.com/13601549/114078874-1192ea00-98dc-11eb-85e5-0743da5dde90.png)

The command I use is:

`python  src/translate.py --residual_velocities --learning_rate 0.005 --iterations 10000 --action all`

I would very appreciate it if you could response me.
